src/main/kotlin/no/skatteetaten/aurora/gobo/resolvers/resolvers.kt

```diff
@@ -15,4 +18,11 @@ val DataFetchingEnvironment.token: String?
             }
             else -> null
         }
-    }
\ No newline at end of file
+    }
+
+fun <T : Any> DataFetchingEnvironment.loader(type: KClass<T>): DataLoader<Any, Try<T>> {
+    val key = "${type.simpleName}DataLoader"
+    val dataLoader = this.getContext<GraphQLContext>().dataLoaderRegistry.get()
+        .getDataLoader<Any, Try<T>>(key) ?: throw IllegalStateException("No $key found")
+    return dataLoader as? NoCacheBatchDataLoader ?: dataLoader as NoCacheBatchDataLoaderFlux
+}
```

Comment:
Dataloader-instansen finnes ved å bruke klassen (typen den er dataloader for), også legger på *DataLoader. Jeg er ikke 100% fornøyd med dette, for det er en indirekte avhengighet her på at dataloaderne må navngis på en viss måte. Syns i hvert fall dette var bedre enn å sende inn en string, men er åpen for andre forslag hvis du kommer på en annen måte å løse dette på.
Skulle også veldig gjerne kunne skrevet dette på denne måten: `dfe.loader<Affiliation>()`
Men det fikk jeg ikke til siden jeg trenger class-objektet runtime.

Reply:
Funker ikke dette?
`inline fun <reified T : Any> DataFetchingEnvironment.loader(): DataLoader<Any, Try<T>> {
    val key = T::class.java.simpleName
    val dataLoader = this.getContext<GraphQLContext>().dataLoaderRegistry.get()
        .getDataLoader<Any, Try<T>>(key) ?: throw IllegalStateException("No $key found")
    return dataLoader as? NoCacheBatchDataLoader ?: dataLoader as NoCacheBatchDataLoaderFlux
}`

Reply:
Se min kommentar i PRen din. Jeg syntes ikke dette er et tilfelle hvor generics gir noe ekstra verdi. Man skal egentlig bare hente ut loader for en gitt klasse. Klassen er parameter. Å gjøre dette generisk gjør egentlig forholdet mellom parameter og returnverdi vanskeligere å forstå. Etter min mening er dette et tilfelle hvor generics ikke bør brukes til å sende typeinformasjon. Dette er ganske enkelt et eksempel på en metode som tilfeldigvis tar en klasse som parameter. Det er ikke noen automatikk i at det dermed er bedre at denne sendes via generics. Et alternativ er å sende navnet på dataloaderen, og ikke typen den er dataloader for.

Reply:
Så lenge man ikke kan hente loader sin type på en reified måte her så er jeg egentlig litt enig i dette. Skal man gjøre dette må alle resolverne få type som CompletableFuture<Try<Affiliation>> f.eks og det er litt mye.